ddtrace/tracer/option.go

```diff
@@ -992,6 +1002,13 @@ func WithProfilerEndpoints(enabled bool) StartOption {
 	}
 }
 
+// WithDebugSpansMode enables or disables the debug mode for logging open spans.
+func WithDebugSpansMode() StartOption {
+	return func(c *config) {
+		c.debugOpenSpans = true
+	}
+}
+
```

Comment:
Let's do something similar to how PartialFlushing is configured. i.e. this should accept the timeout in the form of a `time.Duration`, which will automatically enable the feature and set the timeout. That way it can all be configured at once.

In which case, the godoc would look something like this:
```
// WithDebugSpansMode enables debugging of old spans that have been
// potentially abandoned, which could prevent a trace from being sent to the
// Datadog (especially if partial flushing is off).
// Debug level logging must also be enabled for these logs to appear (e.g.
// by using WithDebugMode or DD_TRACE_DEBUG).
// This can also be configured by setting DD_TRACE_DEBUG_OPEN_SPANS
// to true, which will default to 1 second, unless overridden by DD_TRACE_OPEN_SPAN_TIMEOUT.
// This feature is disabled by default.
```